pkg/pillar/cmd/zedagent/handlecipherconfig.go

```diff
@@ -17,6 +17,28 @@ import (
 
 var cipherCtxHash []byte
 
+// populateCipherContexts: fill in local map from persistent publication on boot
+func populateCipherContexts(ctx *getconfigContext) {
+	items := ctx.pubCipherContext.GetAll()
+	for _, item := range items {
+		context := item.(types.CipherContext)
+		ctx.cipherContexts[context.Key()] = context
+		log.Noticef("populateCipherContext found %s",
+			context.Key())
+	}
+}
+
+// lookupCipherContextByCCH: lookup by ControllerCertHash
+// The Key is a UUID so need to walk map
+func lookupCipherContextByCCH(ctx *getconfigContext, cch []byte) *types.CipherContext {
+	for _, context := range ctx.cipherContexts {
+		if bytes.Equal(context.ControllerCertHash, cch) {
+			return &context
+		}
+	}
+	return nil
+}
```

Comment:
```suggestion
// lookupCipherContextByCCH: lookup by ControllerCertHash
func lookupCipherContextByCCH(ctx *getconfigContext, cch []byte) *types.CipherContext {
	items := ctx.pubCipherContext.GetAll()
	for _, item := range items {
		context := item.(types.CipherContext)
		if bytes.Equal(context.ControllerCertHash, cch) {
			return &context
		}
	}
	return nil
}
```

@eriknordmark maybe I'm missing something, but cannot we just call GetAll from `lookupCipherContextByCCH` removing the need for `populateCipherContexts` and the need to walk the map?

Reply:
Yes, we can but then we might as well get rid of the local cipherContexts map and always use the publication. 
The local map is faster since the pubsub Get/GetAll imply a json encode/decode to get a deep copy of an arbitrary nested struct.

Reply:
yes, it makes sense to delete it: we basically never use it as a map - only walk it

Reply:
I'll do that as a separate PR in master.